### PR TITLE
Fix bugs in keyword definitions

### PR DESCRIPTION
**Pull request by berland**
Uncovered by trying to parse the test suite supplied with Eclipse.

Some missing, some typos, some inconsistencies.

### CODE REVIEW

File: src/opm/parser/eclipse/share/keywords/000_Eclipse100/R/ROCKPAMA [hunk 1]
```diff
@@ -3,6 +3,8 @@
   "sections": [
     "PROPS"
   ],
+  "size": "UNKNOWN",
+  "_comment_size": "It is tempting to use TABDIMS::NTROCC, but that has an untrivial default value",
```
Comment (akva2) [inline src/opm/parser/eclipse/share/keywords/000_Eclipse100/R/ROCKPAMA:7]: non

Comment (berland) [reply src/opm/parser/eclipse/share/keywords/000_Eclipse100/R/ROCKPAMA:7]: ok. Is there a place where we could put this, or should we just hope noone else will insert NTROCC?

Comment (akva2) [reply src/opm/parser/eclipse/share/keywords/000_Eclipse100/R/ROCKPAMA:7]: i don't know but it's nontrivial, not untrivial.

Comment (berland) [reply src/opm/parser/eclipse/share/keywords/000_Eclipse100/R/ROCKPAMA:7]: (yeah, a unit test would do..)

**Comment by joakim-hove** [comment]
Thanks!

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by berland** [comment]
Remaining deficiencies in this particular test suite is parsing `B*` keywords in the `SUMMARY` sections, this seems to be a problem outside of these json files.

**Comment by joakim-hove** [comment]
>Remaining deficiencies in this particular test suite is parsing B* keywords in the SUMMARY sections, this seems to be a problem outside of these json files.

Details details .... ?

**Comment by joakim-hove** [comment]
Have a little more detail about the `AQUFETP` keyword - this has been notoriously difficult to get right; and one of our tests fail with this PR. Might be the test which is faulty though

**Comment by berland** [comment]
> > Remaining deficiencies in this particular test suite is parsing B* keywords in the SUMMARY sections, this seems to be a problem outside of these json files.
> 
> Details details .... ?

This seems to be related to a skipped `/` for every block coordinate for block summary keywords, 

```
BPR
 1 1 1
 2 2 2 
/
```
is accepted by E100, while OPM only accepts the syntax
```
BPR
 1 1 1 /
 2 2 2 /
/
```
where the latter is what is in the documentation.

I guess fixing this is a high-hanging fruit, not suggesting to fix.

**Comment by berland** [comment]
> Have a little more detail about the `AQUFETP` keyword - this has been notoriously difficult to get right; 

The fix in AQUFETP json makes the final `/` in the `AQUFETP` optional, as the number of records is specified in `AQUDIMS`.

```  
AQUDIMS
    0    0    0    0    3  400 /

<snip>

AQUFETP    3 AQUIFERS                   
   1  9000.    1*     .2E+11  .5-05  110    1      / 
   2  1000    1*     .3E+10  .5E-05  50    1      / 
   3  8000    1*     .3E+10  .5E-05  30    2      /
```
The above statements work in E100 but  not OPM. Adding size to AQUFETP json fixes this.

**Comment by joakim-hove** [comment]
> This seems to be related to a skipped / for every block coordinate for block summary keywords,

```
BPR
 1 1 1
 2 2 2 
/
```

OK - that looks like a high-hanging fruit as you say ....

**Comment by joakim-hove** [comment]
> The fix in AQUFETP json makes the final / in the AQUFETP optional, as the number of records is specified in AQUDIMS.

OK - we have been trough several rounds on `AQUFETP` - unfortunately it does not really fit in the model we have in opm-common; the problem is that the size in `AQUDIMS`is only a *maximum* - and if we use the model approach taken here we get problems with `AQUFETP` keywords which are too short - yes we have those as well. So - to really fix the `AQUFETP` keyword for all the situations we have encountered requires changes to the quit-low-level parsing code.

My suggestion is that:
1. You take the `AQUFETP` changes out of this PR - then we can merge it.
2. A new PR is opened with only the `AQUFETP` changes - that will require larger changes before it can be merged; a *high hanging fruit*!

**Comment by berland** [comment]
> 1. You take the `AQUFETP` changes out of this PR - then we can merge it.
> 2. A new PR is opened with only the `AQUFETP` changes - that will require larger changes before it can be merged; a _high hanging fruit_!

I  will go for 1. Supporting optional  closing `/` is not a priority.

**Comment by OPMUSER** [comment]
No fix required as the OPM behavior and documentation is consistent with 
the commercial simulator's documentation, as per:



So the commercial simulator is in error here. So +1 to OPM Flow.

OPM User
opmuser0@gmail.com

**Comment by OPMUSER** [comment]
No fix required as the OPM behavior and documentation is consistent with the commercial simulator's documentation, as per:

amekgcnnaapfikdj


So the commercial simulator is in error here. So +1 to OPM Flow.

**Comment by berland** [comment]
`AQUFET` currently specifies `size` as in this PR.  I assume  `size` should be removed from the json in `AQUFET` also if it should not be in `AQUFETP`.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by OPMUSER** [comment]
No fix required as the OPM behavior and documentation is consistent with the commercial simulator's documentation. For the record I'm referring to the SUMMARY keywords of the form: 

```
BPR
 1 1 1 /
 2 2 2 /
/
```

Not the aquifer keywords.

**Pull request merged**